### PR TITLE
docs(website): add dark theme spec

### PR DESCRIPTION
**Pull request by shleewhite**
## Changes in this PR
* Creates a new section in the website sidebar called Theme
  * Moves `core/theme-package` to `theme/`
  * Moves `core/changing-theme` to `theme/changing-theme`
  * Adds a new page `theme/dark-theme`
 * Remove the Dark Mode FAQ, instead the badge links to the new dark theme page
 * Updates the netlify redirects to reflect the new IA
 * Update URLs on the site for the pages that moved
 * Update Cypress tests to include the new pages in the VRT and sidebar navigation tests

I grouped the changes in commits, so it may be easier to review by looking at each commit.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by gloriliale — APPROVED** [review]
This looks great, @shleewhite!

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarNavigation.tsx [hunk 1]
```diff
@@ -364,6 +358,28 @@ const SidebarNavigation: React.FC = () => {
...
+          level={0}
+          onClick={() =>
+            trackCustomEvent({
+              category: 'Left Navigation',
+              action: 'click-theme',
+              label: 'Theme',
+            })
+          }
+        >
+          <SidebarAnchor level={1} to={`${SidebarCategoryRoutes.THEME}/`}>
```
Comment (TheSisb) [inline packages/paste-website/src/components/site-wrapper/sidebar/SidebarNavigation.tsx:373]:
```suggestion
          <SidebarAnchor level={1} to={`${SidebarCategoryRoutes.THEME}`}>
```

File: packages/paste-website/src/pages/theme/index.mdx [hunk 2]
```diff
@@ -2,15 +2,15 @@
 title: Theme Package
 package: '@twilio-paste/theme'
 description: The Paste Theme Package allows easy and simple access to the design tokens via a theme provider to React components
-slug: /core/theme-package/
+slug: /theme/
```
Comment (TheSisb) [inline packages/paste-website/src/pages/theme/index.mdx:5]:
```suggestion
slug: /theme
```

File: packages/paste-website/netlify.toml [hunk 3]
```diff
@@ -80,13 +80,22 @@
   to = '/components/:splat'
 
 # CORE/
+[[redirects]]
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply packages/paste-website/netlify.toml:83]: :/ this came up when we redid the sidebar. Having it in the paste-website/ directory works locally, but netlify needs the file to be at the root. I think @SiTaggart suggested keeping both for now even tho its not ideal

File: packages/paste-website/src/pages/customization/composing-custom-components-with-design-tokens.mdx [hunk 4]
```diff
@@ -63,7 +63,7 @@ The list of components you're left with may not be expressed via Paste. At this
 
 ## Design Primitive Components
 
-Design Primitive Components are React components that expose style properties using [Styled-System](https://www.styled-system.com). These style properties are typed React props that only expose design tokens as values. There are style properties for each of the categories that are included on the [Theme Object](/core/theme-package#theme) from the [Theme Package](/core/theme-package).
+Design Primitive Components are React components that expose style properties using [Styled-System](https://www.styled-system.com). These style properties are typed React props that only expose design tokens as values. There are style properties for each of the categories that are included on the [Theme Object](/theme/#theme) from the [Theme Package](/theme/).
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply packages/paste-website/src/pages/customization/composing-custom-components-with-design-tokens.mdx:66]: Interesting, I'll update the links!

File: packages/paste-website/netlify.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-website/netlify.toml:83]: Yeah, Netlify CLI which runs the websites locally, needs isolated ones, but Netlify the actual build box, doesn't register the isolated ones and only reads the root. Netlify monorepo support is super lacking, and they know about it.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**